### PR TITLE
feat: account recovery

### PR DESCRIPTION
**Pull request by quadristan**
This MR brings the functionality of using ARK to log in with the CLI, optionally recovering the MP with a **recover-account** command

It will allow users who have issue with their ARK to get more information.

User XP:

```
? Doing a recovery will delete all local data from this app. Continue? yes
? Do you want to see the masterpassword that will be unlocked by this recovery key? yes
✔ The local Dashlane local storage has been reset and you have been logged out.
? Please enter your email address: <SNIPPED>
info: Please open the following URL in your browser:  <SNIPPED>
? Please enter the code you received by email: <SNIPPED>
? Please enter your account recovery key: <SNIPPED>
info: Recovered masterpassword <SNIPPED>
✔ Account recovered! you can use the CLI to export its data.
```

### CODE REVIEW

File: src/modules/crypto/keychainManager.ts [hunk 1]
```diff
@@ -116,7 +129,15 @@
     if (isSSO) {
         masterPassword = decryptSsoRemoteKey({ ssoServerKey, ssoSpKey, remoteKeys });
     } else {
-        masterPassword = masterPasswordEnv ?? (await askMasterPassword());
+        if (recoveryOptions?.promptForArk) {
+            const freshAuthTicket = authTicket ?? (await getAuthenticationTickets(login)).authTicket;
+            masterPassword = await getMasterpasswordFromArkPrompt(freshAuthTicket, login);
+            if (recoveryOptions.displayMasterpassword) {
+                logger.info(`Recovered masterpassword ${masterPassword}`);
```
[the remark this answers is not in the document]
Comment (quadristan) [reply src/modules/crypto/keychainManager.ts:136]: Yes, this is intended for this feature and requires user manual action to display it

**Comment by Mikescops** [comment]
@quadristan 
Some purely functional comments first:
- "you can use the CLI to export its data." you can't do this in the .DASH format via the CLI which would be less useful in this situation, it would be good to link a user support article about the recovery process if possible too
- "Do you want to see the masterpassword that will be unlocked by this recovery key?" i feel like this question can be ask later once they have input everything else

File: src/modules/auth/recovery/getMasterpasswordFromArkPrompt.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    try {
+        const decryptedVaultKey = decryptAesCbcHmac256({
+            cipherData: encryptedData.cipherData,
+            originalKey: derivatedKey,
+            inflatedKey: encryptedData.cipherConfig.cipherMode === 'cbchmac64',
+        });
+        return decryptedVaultKey.toString('utf-8');
+    } catch (e) {
+        logger.error('Failed to decrypt MP. Maybe the ARK entered is incorrect');
+        throw e;
```
Comment (Mikescops) [inline src/modules/auth/recovery/getMasterpasswordFromArkPrompt.ts:32]:
```suggestion
    } catch (error) {
        logger.error('Failed to decrypt Master Password. Maybe the ARK (Account Recovery Key) entered is incorrect');
        throw error;
```

File: src/modules/auth/recovery/getMasterpasswordFromArkPrompt.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import { getEncryptedVaultKey } from '../../../endpoints/getEncryptedVaultKey';
+import { logger } from '../../../logger';
+import { askAccountRecoveryKey } from '../../../utils/dialogs';
+import { decryptAesCbcHmac256, getDerivateUsingParametersFromEncryptedData } from '../../crypto/decrypt';
+import { deserializeEncryptedData } from '../../crypto/encryptedDataDeserialization';
+
+function cleanArk(input: string): string {
```
Comment (Mikescops) [inline src/modules/auth/recovery/getMasterpasswordFromArkPrompt.ts:7]: use arrow function please, here and below

File: src/utils/dialogs.ts [hunk 4]
```diff
@@ -72,6 +72,19 @@ export const askConfirmReset = async () => {
...
+export const askConfirmRecovery = async () => {
+    const response = await confirm({
+        message: 'Doing a recovery will delete all local data from this app. Continue?',
+    });
+    return response;
+};
+
+export const askConfirmShowMp = async () => {
+    const response = await confirm({
+        message: 'Do you want to see the masterpassword that will be unlocked by this recovery key?',
```
Comment (Mikescops) [inline src/utils/dialogs.ts:84]:
```suggestion
        message: 'Do you want to see the master password that will be unlocked by this recovery key?',
```

**Pull request closed without merging**